roles/falcon_installation/tasks/api.yml

```
@@ -22,33 +22,25 @@
       Content-Type: application/json
   register: falcon_api_target_cid
 
-- name: CrowdStrike Falcon | Get list of all Falcon sensors
-  uri:
-    url: "https://{{ falcon_cloud }}/sensors/combined/installers/v1?filter=platform%3A%22{{ falcon_os_family }}*%22%2Bos%3A%22{{ falcon_target_os }}%22&=&sort=version.desc"
-    method: GET
-    return_content: true
-    headers:
-      authorization: "Bearer {{ falcon_api_oauth2_token.json.access_token }}"
-  register: falcon_api_installer_list
-
-- name: CrowdStrike Falcon | Get SHA256 hash of Falcon installer for the required OS Version
+- name: "CrowdStrike Falcon | Build API Query"
   set_fact:
-    falcon_api_sha_hash: "{{ falcon_api_installer_list.json.resources | selectattr('os_version', 'contains', ansible_distribution_major_version ) }}"
+    falcon_os_query: "{{ 'os:\"' + falcon_target_os + '\"+os_version:\"' + falcon_os_version + '\"' }}"
 
-- name: CrowdStrike Falcon | Detect Name of Latest Falcon Sensor
+- name: CrowdStrike Falcon | Get list of all Falcon sensors
   uri:
-    url: "https://{{ falcon_cloud }}/sensors/combined/installers/v1?filter=platform%3A%22{{ falcon_os_family }}*%22%2Bos%3A%22{{ falcon_target_os }}%22&=&sort=version.desc&limit=1"
+    url: "https://{{ falcon_cloud }}/sensors/combined/installers/v1?filter={{ falcon_os_query | urlencode }}"
     method: GET
     return_content: true
     headers:
       authorization: "Bearer {{ falcon_api_oauth2_token.json.access_token }}"
-  register: falcon_api_installer_version
+      Content-Type: application/json
+  register: falcon_api_installer_list
 
 - name: CrowdStrike Falcon | Download Latest Sensor Installation Package
```

```suggestion
- name: CrowdStrike Falcon | Download Falcon Sensor Installation Package
```
I would remove latest since if they opt for n-N, then it isn't the latest

---

Done